src/SelfServe/SelfServeUtils.test.tsx

```diff
@@ -4,35 +4,70 @@ import {
   SelfServeBaseClass,
   updateContextWithDecorator,
 } from "./SelfServeUtils";
-import { InputType, UiType } from "./../Explorer/Controls/SmartUi/SmartUiComponent";
+import { NumberUiType, SmartUiInput } from "./../Explorer/Controls/SmartUi/SmartUiComponent";
+import { RefreshResult, SelfServeNotification } from "./SelfServeComponent";
 
 describe("SelfServeUtils", () => {
   it("initialize should be declared for self serve classes", () => {
     class Test extends SelfServeBaseClass {
-      public onSubmit = async (): Promise<void> => {
-        return;
+      public validate: (currentvalues: Map<string, SmartUiInput>) => string;
+      public onRefresh: () => Promise<RefreshResult>;
+      public onSave = async (): Promise<SelfServeNotification> => {
```

Comment:
This works. It might be easier to do something like: `public onSave = jest.fn()`.